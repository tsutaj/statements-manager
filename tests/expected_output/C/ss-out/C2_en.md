


## Problem C2

### <u>**Static Range Sum**</u>

### Problem Statement

You are given a non-negative integer sequence $A = (a_0, a_1, \ldots, a_{N-1})$ with the length $N$. Process the following $Q$ queries in order:

- You are given integers $l_i$​ and $r_i$​. Print $\sum_{k=l_i}^{r_i-1} a_k$​.

### Note

The difference between C1 and C2 is only constraints of $N$ and $Q$.

### Input

```
$N$ $Q$
$a_0$​ $a_1$​ $\ldots$ $a_{N-1}$​
$l_1$​ $r_1$​
$\vdots$
$l_Q$​ $r_Q$​
```

- All inputs are integers.
- $1 \leq N \leq 100\\,000$
- $1 \leq Q \leq 100\\,000$
- $-10^{9} \leq a_i \leq 10^{9}$
- $1 \leq l_i < r_i \leq N$

### Sample


### Sample Input 
```
5 5
1 10 100 1000 10000
2 3
0 3
2 5
3 4
0 5
```



### Sample Output 
```
100
111
11100
1000
11111
```








