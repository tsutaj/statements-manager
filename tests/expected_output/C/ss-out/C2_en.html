<!DOCTYPE html>
<html lang="ja">

<head>
    <title>Problem Statement</title>
    <meta charset="utf-8">
    <link href='http://fonts.googleapis.com/css?family=Noto+Serif' rel='stylesheet' type='text/css'>
    <style type="text/css">
        html {
            color: #222222;
            font-family: "Noto Serif";
        }

        h1 {
            font-size: 16pt;
            border-bottom: solid 2px #e0e0e0;
            margin-bottom: 20px;
        }

        h2 {
            font-size: 14pt;
            border-bottom: solid 2px #e0e0e0;
            margin-bottom: 20px;
            margin-top: 20px;
        }

        h3 {
            font-size: 14pt;
            border-bottom: solid 2px #e0e0e0;
            margin-bottom: 20px;
            margin-top: 20px;
        }

        h4 {
            font-size: 13pt;
            margin-bottom: 20px;
            margin-top: 20px;
        }

        h5 {
            font-size: 12pt;
            margin-bottom: 20px;
            margin-top: 20px;
        }

        b {
            font-weight: bold;
        }

        p {
            font-size: 12pt;
        }

        li {
            font-size: 12pt;
        }

        pre {
            font-size: 11pt;
            padding: 8px;
            border: solid 1px #e0e0e0;
            border-radius: 4px;
        }

        code {
            color: #e83e8c;
        }

        code.language-input-format {
            color: #222222;
        }

        .testcase_pre {
            font-size: 11pt;
            padding: 8px;
            border: solid 1px #e0e0e0;
            border-radius: 4px;
            overflow: auto;
            max-width: 480;
            max-height: 340;
        }

        .error_pre {
            font-size: 11pt;
            padding: 8px;
            border: solid 1px #f0e0e0;
            border-radius: 4px;
            overflow: auto;
            max-height: 340;
            background-color: #ffeeee;
        }

        .table tbody>tr>td.vert-aligned {
            vertical-align: middle;
        }
    </style>
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({
            tex2jax: {
                inlineMath: [["$", "$"]],
                processEscapes: true,
                processClass: 'input-format|language-input-format'
            }
        });
    </script>
    <script type="text/javascript"
        src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js?config=TeX-AMS_CHTML">
        </script>
</head>

<body>
    
    <div >
        <h2>Problem C2</h2>
<h3><u><strong>Static Range Sum</strong></u></h3>
<h3>Problem Statement</h3>
<p>You are given a non-negative integer sequence $A = (a_0, a_1, \ldots, a_{N-1})$ with the length $N$. Process the following $Q$ queries in order:</p>
<ul>
<li>You are given integers $l_i$​ and $r_i$​. Print $\sum_{k=l_i}^{r_i-1} a_k$​.</li>
</ul>
<h3>Note</h3>
<p>The difference between C1 and C2 is only constraints of $N$ and $Q$.</p>
<h3>Input</h3>
<pre class="input-format"><code class="language-input-format">$N$ $Q$
$a_0$​ $a_1$​ $\ldots$ $a_{N-1}$​
$l_1$​ $r_1$​
$\vdots$
$l_Q$​ $r_Q$​
</code></pre>
<ul>
<li>All inputs are integers.</li>
<li>$1 \leq N \leq 100\,000$</li>
<li>$1 \leq Q \leq 100\,000$</li>
<li>$-10^{9} \leq a_i \leq 10^{9}$</li>
<li>$1 \leq l_i &lt; r_i \leq N$</li>
</ul>
<h3>Sample</h3>
<h3>Sample Input</h3>
<pre class="input-format"><code class="language-input-format">5 5
1 10 100 1000 10000
2 3
0 3
2 5
3 4
0 5
</code></pre>
<h3>Sample Output</h3>
<pre class="input-format"><code class="language-input-format">100
111
11100
1000
11111
</code></pre>
    </div>
    
</body>

</html>